### PR TITLE
feat: modify minor compaction to make each ColumnFile in persisted set is sorted

### PR DESCRIPTION
**Pull request by hongyunyan**
### What problem does this PR solve?

Issue Number: ref #5252 

Problem Summary:

### What is changed and how it works?

Modify minor compaction to make each ColumnFile in persisted set is sorted, thus we can speed up reading under fast mode. Besides, a more ordered delta value space will also improve read performance in normal mode.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

test by dtworkload， environment: M1 Pro 32G
image

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by hongyunyan** [comment]
cc @lidezhu @flowbehappy

File: dbms/src/Storages/DeltaMerge/Delta/DeltaValueSpace.cpp
Comment (lidezhu) [inline dbms/src/Storages/DeltaMerge/Delta/DeltaValueSpace.cpp:266]:
```suggestion
void DeltaValueSpace::updateDeltaTree(const DeltaIndex::Updates & delta_index_updates)
```

File: dbms/src/Storages/DeltaMerge/DeltaMergeStore.cpp
Comment (lidezhu) [inline dbms/src/Storages/DeltaMerge/DeltaMergeStore.cpp:1611]: After `compactDelta`, it may be not a good point to `placeDeltaIndex`.  
Because after flush task complete, it indicates that there may some fresh data at the tail which just flush from memory to disk. So it's a good signal to try to `placeDeltaIndex`. 
But `compactDelta` is just a reorganization of the old data, so I suggest not adding this line here.

**Review by lidezhu — APPROVED** [review]
LGTM

**Comment by flowbehappy** [comment]
/hold
There are some performance regressions. Should be figured out before merging.

**Pull request closed without merging**
